velox/exec/trace/QueryDataWriter.cpp

```diff
@@ -45,22 +53,33 @@ void QueryDataWriter::write(const RowVectorPtr& rows) {
   batch_->append(rows);
   dataType_ = rows->type();
 
+  const auto fileSize = dataFile_->size();
   // Serialize and write out each batch.
   IOBufOutputStream out(
       *pool_, nullptr, std::max<int64_t>(64 * 1024, batch_->size()));
   batch_->flush(&out);
   batch_->clear();
   auto iobuf = out.getIOBuf();
   dataFile_->append(std::move(iobuf));
+  if (FOLLY_UNLIKELY(
+          updateAndCheckTraceLimitCB_(dataFile_->size() - fileSize))) {
+    exceedTraceLimit_ = true;
+    finish();
```

Comment:
We probably don't need exceedTraceLimit_ and let finish() takes limitExceeded flag? Thanks!